File: 0135-candy/0135-candy.c
// 計算最少需要多少糖果
int candy(int* ratings, int ratingsSize) {
    if (ratingsSize == 0) return 0;

    // 動態配置一個陣列來記錄每個孩子的糖果數，初始化為 1
    int* candies = (int*)malloc(sizeof(int) * ratingsSize);
    for (int i = 0; i < ratingsSize; i++) {
        candies[i] = 1;
    }

    // 從左到右掃描
    for (int i = 1; i < ratingsSize; i++) {
        if (ratings[i] > ratings[i - 1]) {
            candies[i] = candies[i - 1] + 1;
        }
    }

    // 從右到左掃描
    for (int i = ratingsSize - 2; i >= 0; i--) {
        if (ratings[i] > ratings[i + 1]) {
            if (candies[i] < candies[i + 1] + 1) {
                candies[i] = candies[i + 1] + 1;
            }
        }
    }

    // 計算總和
    int total = 0;
    for (int i = 0; i < ratingsSize; i++) {
        total += candies[i];
    }

    free(candies);  // 記得釋放記憶體
    return total;
}

File: 0135-candy/0135-candy.py
class Solution:
    def candy(self, ratings: List[int]) -> int:
        n = len(ratings)
        candies = [1] * n

        # 從左到右
        for i in range(1, n):
            if ratings[i] > ratings[i - 1]:
                candies[i] = candies[i - 1] + 1

        # 從右到左
        for i in range(n - 2, -1, -1):
            if ratings[i] > ratings[i + 1]:
                candies[i] = max(candies[i], candies[i + 1] + 1)

        return sum(candies)